web/api/formatters/json_wrapper.c

```diff
@@ -132,43 +124,54 @@ void rrdr_json_wrapper_begin(RRDR *r, BUFFER *wb, uint32_t format, RRDR_OPTIONS
     }
     buffer_strcat(wb, "],\n");
 
-    if (rrdset_query_data->show_dimensions) {
+    if (r->internal.query_options & RRDR_OPTION_ALL_DIMENSIONS) {
         buffer_sprintf(wb, "   %sfull_dimension_list%s: [", kq, kq);
 
         char name[RRD_ID_LENGTH_MAX * 2 + 2];
         char output[RRD_ID_LENGTH_MAX * 2 + 8];
 
         struct value_output co = {.c = 0, .wb = wb};
 
-        DICTIONARY *dict = dictionary_create(DICT_OPTION_SINGLE_THREADED);
-        for (i = 0, rd = temp_rd ? temp_rd : r->st->dimensions; rd; rd = rd->next) {
-            snprintfz(name, RRD_ID_LENGTH_MAX * 2, "%s:%s", rrddim_id(rd), rrddim_name(rd));
-            int len = snprintfz(output, RRD_ID_LENGTH_MAX * 2 + 7, "[\"%s\",\"%s\"]", rrddim_id(rd), rrddim_name(rd));
-            dictionary_set(dict, name, output, len+1);
+        DICTIONARY *dict = dictionary_create(DICT_OPTION_SINGLE_THREADED|DICT_OPTION_DONT_OVERWRITE_VALUE);
+        for (c = 0; c < qt->metrics.used ;c++) {
```

Comment:
From `32bits armvl7` we have this warning:

````sh
web/api/formatters/json_wrapper.c: In function 'rrdr_json_wrapper_begin':
web/api/formatters/json_wrapper.c:136:23: warning: comparison of integer expressions of different signedness: 'long int' and 'uint32_t' {aka 'unsigned int'} [-Wsign-compare]
  136 |         for (c = 0; c < qt->metrics.used ;c++) {
      |                       ^
web/api/formatters/json_wrapper.c:153:23: warning: comparison of integer expressions of different signedness: 'long int' and 'uint32_t' {aka 'unsigned int'} [-Wsign-compare]
  153 |         for (c = 0; c < qt->instances.used ; c++) {
      |                       ^
web/api/formatters/json_wrapper.c:172:23: warning: comparison of integer expressions of different signedness: 'long int' and 'uint32_t' {aka 'unsigned int'} [-Wsign-compare]
  172 |         for (c = 0; c < qt->instances.used ; c++) {

```

Reply:
solved